src/main/java/algorithms/orthogonalRangeSearching/oneDim/README.md

```diff
@@ -0,0 +1,54 @@
+# 1D Orthogonal Range Searching
+
+1D orthogonal range searching is a computational problem where you search for elements or data points within a 
+specified 1D range (interval) in a collection of 1D data (e.g. Find me everyone between ages 22 and 27). 
+
+While we can simply sort (and binary search the low and high of the specified range) to return all data points within 
+the specified range, each query will minimally take O(nlogn) due to the sorting step, assuming dynamic updating of 
+the data.
+
+The goal of 1D Orthogonal Range Searching is to efficiently identify and retrieve all data points that fall within the 
+given range.
+
+Strategy:
+1. Use a binary search tree
+2. Store all points in the leaves of the tree (internal nodes only store copies)
+3. Each internal node v stores the MAX of any leaf in the left sub-tree (**range tree property**)
+
+![1DORS](../../../../../../docs/assets/images/1DORS.jpg)
+
+Say we want to find all the nodes between 10 and 50 i.e. query(10, 50). We would want to:
+1. Find split node: highest node where search includes both left & right subtrees
+=> we want to make use of the range tree property to perform binary search to find our split node. See findSplit(root, low, high) in code. 
+2. Left traversal & right traversal
+
+![1DORSQuery](../../../../../../docs/assets/images/1DORSQuery.jpeg)
+Image Source: Lecture Slides
+
+## Complexity Analysis
+**Time**:
+
+Build Tree (cost incurred once only): O(nlogn) limited by sorting step
```

Comment:
could you also provide the complexity for building the tree. In practice, if the user provides a sorted array, a O(n) validation check would suffice. No need to tweak the code further, can just mention this.